### ISSUE TITLE
Vite plugin not working with monorepo on Windows

### ISSUE THREAD

**Issue by mc-petry** [vanilla-extract-css/vanilla-extract#780]
### Describe the bug

**@vanilla-extract/vite-plugin@3.3.0** not working in monorepo setup.
Oldest version **3.1.4** works.

It seems that
```tsx
  const getAbsoluteVirtualFileId = (source: string) => normalizePath(path.join(config.root, source));
```
not working in a right way on windows os.

Fast change to:
```tsx
  const getAbsoluteVirtualFileId = (source: string) => source.startsWith('..')
    ? normalizePath(path.join(config.root, source))
    : source.startsWith('/@id/') ? source.substring(5) : source
```
fixes build but breaks HMR.

### Reproduction

https://github.com/mc-petry/vanilla-extract-monorepo/tree/vite-plugin-regression

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 AMD Ryzen 3 5300U with Radeon Graphics
    Memory: 2.24 GB / 15.35 GB
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    npm: 8.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (103.0.1264.77)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @vanilla-extract/vite-plugin: 3.3.0 => 3.3.0
    vite: ^3.0.5 => 3.0.5
```

### Used Package Manager

pnpm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/seek-oss/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by riccardoperra** [comment]
I have the same issue, still not updating since 3.1.4 is the latest version that works when you have to bundle .css.ts files from the app side. 

Is the /@id/ check related of how pnpm resolve the packages paths 🤔?

**Comment by TomQuast** [comment]
I had the same problem running 3.2.7.
Just upgraded to the latest version and it seems fixed :)

**Comment by askoufis** [comment]
This seems to be fixed as of `vite@~4.3.0`. It even works with v3.2.0 of the VE vite plugin.
